Sources/StreamChat/Utils/MessagesPaginationStateHandling/MessagesPaginationState.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The current state of the messages pagination.
+struct MessagesPaginationState {
+    // MARK: State
+
+    /// The oldest fetched message while paginating.
+    var oldestFetchedMessage: MessagePayload?
+    /// The newest fetched message while paginating.
+    var newestFetchedMessage: MessagePayload?
+
+    /// A Boolean value that returns whether the newest messages have all been loaded or not.
+    /// If false, it means that the channel is currently in a mid-page.
+    var hasLoadedAllNextMessages: Bool
+    /// A Boolean value that returns whether the oldest messages have all been loaded or not.
+    var hasLoadedAllPreviousMessages: Bool
+
+    /// A Boolean value that returns whether the channel is currently loading next (new) messages.
+    var isLoadingNextMessages: Bool
+    /// A Boolean value that returns whether the channel is currently loading previous (old) messages.
+    var isLoadingPreviousMessages: Bool
+    /// A Boolean value that returns whether the channel is currently loading a page around a message.
+    var isLoadingMiddleMessages: Bool
+
+    // MARK: Computed Properties
+
+    /// A Boolean value that returns whether the channel is currently loading messages, either old/new or mid pages.
```

Comment:
```suggestion
    /// A Boolean value that returns whether the channel is currently loading messages on either previous, mid or next pages.
```
Better to remain consistent to the property naming and prefer previous/next over old/new.